lib/symlink/index.js

```diff
@@ -51,22 +51,24 @@ function symlink(outFolder, opt) {
       srcPath = path.relative(file.base, srcPath);
     }
 
-    // TODO: make DRY with .dest()
-    var dirMode = number(opt.dirMode, file);
-    var writeFolder = path.dirname(file.path);
-
-    fo.mkdirp(writeFolder, dirMode, onMkdirp);
+    // Because fs.symlink does not allow atomic overwrite option with flags, we
+    // delete and recreate if the link already exists and overwrite is true.
+    if (file.flag === 'w') {
+      // TODO What happens when we call unlink with windows junctions?
+      fs.unlink(file.path, onUnlink);
+    } else {
+      fs.symlink(srcPath, file.path, symType, onSymlink);
+    }
 
-    function onMkdirp(mkdirpErr) {
-      if (mkdirpErr) {
-        return callback(mkdirpErr);
+    function onUnlink(unlinkErr) {
+      if (fo.isUnlinkError(unlinkErr)) {
+        return callback(unlinkErr);
       }
-
       fs.symlink(srcPath, file.path, symType, onSymlink);
     }
 
     function onSymlink(symlinkErr) {
-      if (isErrorFatal(symlinkErr)) {
+      if (fo.isOverwriteError(symlinkErr, file.flag)) {
```

Comment:
This option will need to be added to the documentation (copy-paste from `.dest` options).